docs/specification/draft/client/elicitation.mdx

```diff
@@ -321,12 +526,212 @@ Servers should handle each state appropriately:
+    Note over Server: Confirm: elicitation matches user session<br>Confirm: user is logged into MCP Server or MCP AS
+    Server->>3AS: Exchange authorization code for  OAuth tokens
+    3AS->>Server: Grants tokens
+    Note over Server: Bind tokens to MCP user identity
+    Server-->>Client: elicitation/track response (optional)
+    Client->>Server: Retry tools/call
+    Note over Server: Retrieve token bound to user identity
+    Server->>3RS: Call 3rd-party API
+```
+
+This pattern maintains clear security boundaries while enabling rich integrations with third-party services that require user authorization.
+
+## Error Handling
+
+Servers **SHOULD** return standard JSON-RPC errors for common failure cases:
+
+- Unknown `elicitationId` in an `elicitation/track` request: `-32602` (Invalid params)
+- An `elicitationId` belonging to a different client in an `elicitation/track` request: `-32602` (Invalid params)
+- When a request cannot be processed until an elicitation is completed: `-32604` (ElicitationRequired)
+
+Clients **SHOULD** return standard JSON-RPC errors for common failure cases:
+
+- Server sends an `elicitation/create` request with a mode not declared in client capabilities: `-32602` (Invalid params)
+
 ## Security Considerations
 
-1. Servers **MUST NOT** request sensitive information through elicitation
+1. Clients **MUST** provide clear indication of which server is requesting information
 2. Clients **SHOULD** implement user approval controls
-3. Both parties **SHOULD** validate elicitation content against the provided schema
-4. Clients **SHOULD** provide clear indication of which server is requesting information
-5. Clients **SHOULD** allow users to decline elicitation requests at any time
-6. Clients **SHOULD** implement rate limiting
-7. Clients **SHOULD** present elicitation requests in a way that makes it clear what information is being requested and why
+3. Clients **SHOULD** allow users to decline elicitation requests at any time
+4. Clients **SHOULD** implement rate limiting
+5. Clients **SHOULD** present elicitation requests in a way that makes it clear what information is being requested and why
+6. Servers **MUST** bind elicitation requests to the user's identity
+
+### Safe URL Handling
+
+Clients implementing url elicitation **MUST** handle URLs carefully to prevent users from unknowingly clicking malicious links.
+
+1. Servers **MUST NOT** include URLs in any message or schema fields as part of a form elicitation request.
+2. Servers **MUST NOT** include URLs in any message or schema fields as part of an url elicitation request, except for the `url` field.
+
+These requirements ensure that client implementations have clear rules about when to present a URL to the user, so that other rules about user consent and SSRF protection (below) can be consistently applied.
+
+### Identifying the User
+
+Servers **MUST NOT** rely on client-provided user identification, as this can be forged.
```

Comment:
What does this mean in practice?

Reply:
The server must not trust the user (or MCP client) saying "This is bill@microsoft.com". Only the AS can be trusted to identify the user.

Reply:
@ciamshrek You nailed it, the intent is precisely to communicate "never trust the client". We updated the language here.